dockyard/common/link.py

```diff
@@ -0,0 +1,20 @@
+
+def make_url(host='127.0.0.1', port=None, protocol='http', url=None):
```

Comment:
I expect method like to support large number of options, this function does not seems to be flexible enough to support all the parameter that we can receive or send in REST APIs.